src/types.jl

```diff
@@ -1,6 +1,15 @@
 const DEFAULT_DIM_TYPE = FixedRational{Int32, 2^4 * 3^2 * 5^2 * 7}
 const DEFAULT_VALUE_TYPE = Float64
 
+abstract type AbstractQuantity{T,R} end
+abstract type AbstractDimensions{R} end
+
+constructor_of(::Type{D}) where {D<:AbstractDimensions} = D
+constructor_of(::Type{D}) where {R,D<:AbstractDimensions{R}} = D.name.wrapper
+constructor_of(::Type{Q}) where {Q<:AbstractQuantity} = Q
+constructor_of(::Type{Q}) where {T,Q<:AbstractQuantity{T}} = Q.body.name.wrapper
+constructor_of(::Type{Q}) where {T,R,Q<:AbstractQuantity{T,R}} = Q.name.wrapper
```

Comment:
The tricky thing about these is they break if the user adds another type parameter. Doesn't seem like there's a general way to go from, e.g.,
```julia
AbstractArray
AbstractArray{T}
AbstractArray{T,2}
```

to `AbstractArray`, in a way that accounts for any number of type parameters.